src/pages/challenges.tsx

```diff
@@ -30,58 +36,57 @@ const Challenges: FC = () => {
   }, []);
 
   useEffect(() => {
-    if ('challenges' in response) {
-      let challenges: challenge['content'] = response.challenges;
-      const challengeMap: any = [];
-
-      challengeMap.Active = challenges.filter((task) => task.is_active);
-      challengeMap.Completed = challenges.filter((task) => !task.is_active);
-
-      setFilteredChallenge(challengeMap);
+    if (isLoggedIn && !Object.keys(response).length) {
+      callAPI();
+      if ('challenges' in response) {
+        let challenges: challenge['content'] = response.challenges;
+        const challengeMap: any = [];
+        challengeMap.Active = challenges.filter((task) => task.is_active);
+        challengeMap.Completed = challenges.filter((task) => !task.is_active);
```

Comment:
How are we handling blocked or stalled tasks here?
It looks like the challenge is either active or completed in above case

Reply:
I didn't make this change, I just added it to a condition

Reply:
Ok, so it is tested right?
All the functionalities are meeting expectations

Reply:
The test cases for this hasn't been written yet